# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/udev/hotfix-nios2.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- systemd-258/src/include/override/sys/syscall.h.vanilla	2025-10-02 11:19:43.351529400 +0200
+++ systemd-258/src/include/override/sys/syscall.h	2025-10-02 11:21:23.925182576 +0200
@@ -291,6 +291,8 @@
 #    else
 #      error "Unknown MIPS ABI"
 #    endif
+#  elif defined(__nios2__)
+#    define systemd_NR_open_tree_attr 467
 #  elif defined(__hppa__)
 #    define systemd_NR_open_tree_attr 467
 #  elif defined(__powerpc__)
@@ -501,6 +503,8 @@
 #    else
 #      error "Unknown MIPS ABI"
 #    endif
+#  elif defined(__nios2__)
+#    define systemd_NR_removexattrat 466
 #  elif defined(__hppa__)
 #    define systemd_NR_removexattrat 466
 #  elif defined(__powerpc__)
@@ -571,6 +575,8 @@
 #    else
 #      error "Unknown MIPS ABI"
 #    endif
+#  elif defined(__nios2__)
+#    define systemd_NR_setxattrat 463
 #  elif defined(__hppa__)
 #    define systemd_NR_setxattrat 463
 #  elif defined(__powerpc__)
